app/models/mixins/relationship_mixin.rb

```diff
@@ -527,10 +581,14 @@ def child_types(*args)
   end
 
   def add_parent(parent)
+    return parent.add_children(self) if use_ancestry?
```

Comment:
```suggestion
    return update(:parent => parent) if use_ancestry?
```

they are semantically the same, but you are telling this object to set its parent. this is straight forward in ancestry, lets leverage that.

Reply:
double check the usage of `add_parent` and see if the object is saved or not.
think with relationships everything had to be saved before adding a relationship.
If I remember correctly, this was a difficulty in the past